### ISSUE TITLE
Object properties in data models

### ISSUE THREAD

**Issue by dumboster** [cofoundry-cms/cofoundry#267]
Hello,
Is there a way to add object properties in data models ? I need this in order to subdivide properties on data model and improve the readability of HMI with artificial sections.
A the moment, I'm using [NestedDataModelCollection(MinItems = 1, MaxItems = 1)] but it is not fully appropriate.
That would be great if we had sections like nested collections with a header title and the content, perhaps "collapsible".
Thanx

**Comment by HeyJoel** [comment]
Do you mean grouping data model properties by section? That's not implemented yet, [#58 covers that](https://github.com/cofoundry-cms/cofoundry/issues/58)

**Comment by dumboster** [comment]
Yes, but if possible, I would like to group properties in a class.

**Comment by HeyJoel** [comment]
I see, so something like this?

```csharp
public class ProductDataModel : ICustomEntityDataModel
{
    [PropertyGroup(Collapsable = false)]
    public ProductSummaryData Summary { get; set; }

    [PropertyGroup]
    [Display(Description = "Search engine optimization properties")]
    public SEOData SEO { get; set; }

    [PropertyGroup]
    public ProductPricingData Pricing { get; set; }
}
```

And the editor would render each nested class in a separate section?

**Comment by dumboster** [comment]
Yes exactly!
